### PR TITLE
SDL remote control baseline

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
HMI support implementation of https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0071-remote-control-baseline.md

### CODE REVIEW

File: app/StateManager.js [hunk 1]
```diff
@@ -202,92 +213,71 @@ var StateManager = Em.StateManager.extend(
...
+                enter: function() {
+                  SDL.MediaController.set('activeState',
+                    SDL.States.nextState);
+                  if (!SDL.RadioModel.radioControlStruct.radioEnable) {
+                    SDL.RadioModel.radioEnableKeyPress();
+                  }
+                  this._super();
+                },
                 exit: function() {
+                  //SDL.MediaController.deactivateRadio();
```
Comment (yang1070) [inline app/StateManager.js]: please remove the commented out code

File: app/StateManager.js [hunk 2]
```diff
@@ -202,92 +213,71 @@ var StateManager = Em.StateManager.extend(
...
-                exit: function() {
+                enter: function() {
+                  SDL.MediaController.set('activeState',
+                    SDL.States.nextState);
                   this._super();
+                },
+                exit: function() {
+                  SDL.MediaController.deactivateRadio();
+                  //SDL.MediaController.deactivateCD();
+                  //SDL.MediaController.currentSelectedPlayer.pause();
```
Comment (yang1070) [inline app/StateManager.js]: if the code is not used, please remove

File: app/model/media/RadioModel.js [hunk 3]
```diff
@@ -279,14 +317,30 @@ SDL.RadioModel = Em.Object.create({
           'duration': 10
         }
       },
-      '953': {
+      '650': {
         'radioStation': {
           'frequency': 95,
           'fraction': 3,
           'availableHDs': 0,
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: 650 should have 
frequency': 65,
            'fraction': 0,

File: app/model/media/RadioModel.js [hunk 4]
```diff
@@ -279,14 +317,30 @@ SDL.RadioModel = Em.Object.create({
...
+          'album': 'Album2',
+          'year': 2002,
+          'duration': 20
+        }
+      },
+      '800': {
+        'radioStation': {
+          'frequency': 100,
+          'fraction': 1,
+          'availableHDs': 0,
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: 800 should have 
+          'frequency': 800,
 +          'fraction': 0,

File: app/model/media/RadioModel.js [hunk 5]
```diff
@@ -295,7 +349,7 @@ SDL.RadioModel = Em.Object.create({
           'duration': 30
         }
       },
-      '1035': {
+      '1095': {
         'radioStation': {
           'frequency': 103,
           'fraction': 5,
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: FM109.5 shoul have 
'frequency': 109,

File: app/model/media/RadioModel.js [hunk 6]
```diff
@@ -295,7 +349,7 @@ SDL.RadioModel = Em.Object.create({
           'duration': 30
         }
       },
-      '1035': {
+      '1095': {
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: AM 1095 should have 
frequency': 1095,
'fraction': 0

File: app/model/media/RadioModel.js [hunk 7]
```diff
@@ -311,7 +365,7 @@ SDL.RadioModel = Em.Object.create({
           'duration': 40
         }
       },
-      '1053': {
+      '1450': {
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: AM1450 should have
           'frequency': 1450,
            'fraction': 0

File: app/model/media/RadioModel.js [hunk 8]
```diff
@@ -399,79 +498,294 @@ SDL.RadioModel = Em.Object.create({
...
+        if (this.radioControlStruct.band == 'FM') {
+          properties.push('frequencyInteger');
+          properties.push('frequencyFraction');
+        }
+        if (this.radioControlStruct.band == 'AM') {
+          properties.push('frequencyInteger');
+        }
+        if (this.radioControlStruct.band == 'XM') {
+          properties.push('hdChannel');
+          properties.push('availableHDs');
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: as we discussed offline hdChannel and availableHDs shall be in FM section if HDRaido is available. XM is not related to it. and totally different story.

File: app/model/media/RadioModel.js [hunk 9]
```diff
@@ -518,32 +830,49 @@ SDL.RadioModel = Em.Object.create({
...
+      this.set('station',
+        this.radioControlStruct.frequencyInteger + '.' +
+        this.radioControlStruct.frequencyFraction
+      );
+    }
+    if (this.radioControlStruct.band === 'AM') {
+      this.set('station', this.radioControlStruct.frequencyInteger);
+    }
+    if (this.radioControlStruct.band === 'XM') {
+      this.set('station', 'Channel ' + this.radioControlStruct.hdChannel);
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: hdChannel is not bind to XM

File: app/model/media/RadioModel.js [hunk 10]
```diff
@@ -553,40 +882,44 @@ SDL.RadioModel = Em.Object.create({
...
-      );
+      if (this.radioControlStruct.band === 'FM') {
+        this.setFrequencyInteger(parseInt(data.slice(0, -1)));
+        this.setFrequencyFraction(parseInt(data.slice(-1)));
+      }
+      if (this.radioControlStruct.band === 'AM') {
+        this.setFrequencyInteger(parseInt(data));
+      }
+      if (this.radioControlStruct.band === 'XM') {
+        this.setCurrentHdChannel(parseInt(data.slice(-1)))
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: same here

File: app/model/media/RadioModel.js [hunk 11]
```diff
@@ -765,32 +1153,225 @@ SDL.RadioModel = Em.Object.create({
...
+        properties.push('frequencyInteger');
+        properties.push('frequencyFraction');
+        this.switchRadioBandFrequency(true);
+      }
+      if (this.radioControlStruct.band == 'AM') {
+        properties.push('frequencyInteger');
+        this.switchRadioBandFrequency(true);
+      }
+      if (this.radioControlStruct.band == 'XM') {
+        properties.push('hdChannel');
```
Comment (yang1070) [inline app/model/media/RadioModel.js]: same here

File: app/StateManager.js (hunk 1 above)
Comment (LuxoftAKutsan) [inline app/StateManager.js]: @yang1070 fixed 5fda6adbceb1fce9dbb3d6374031dec65fe37297

File: app/StateManager.js (hunk 2 above)
Comment (LuxoftAKutsan) [inline app/StateManager.js]: @yang1070 fixed 5fda6adbceb1fce9dbb3d6374031dec65fe37297

File: app/model/media/RadioModel.js (hunk 3 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fixed cb9605f619ccfd79c8cbfbcc00002b63b3c0b740

File: app/model/media/RadioModel.js (hunk 4 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fixed cb9605f619ccfd79c8cbfbcc00002b63b3c0b740

File: app/model/media/RadioModel.js (hunk 6 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fixed cb9605f619ccfd79c8cbfbcc00002b63b3c0b740

File: app/model/media/RadioModel.js (hunk 7 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fixed cb9605f619ccfd79c8cbfbcc00002b63b3c0b740

File: app/model/media/RadioModel.js (hunk 8 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fix available 0a8c3fe

File: app/model/media/RadioModel.js (hunk 9 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fix available 0a8c3fe

File: app/model/media/RadioModel.js (hunk 10 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fix available 0a8c3fe

File: app/model/media/RadioModel.js (hunk 11 above)
Comment (LuxoftAKutsan) [inline app/model/media/RadioModel.js]: @yang1070 fix available 0a8c3fe

**Pull request merged**